sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -255,15 +262,20 @@ class ProjectOverviewView extends VerticalLayout{
 
 
     private void tryToDownloadManifest() {
+        Optional<ProjectSummary> selectedSummary = Optional.empty()
         try {
-            Optional.ofNullable(viewModel.selectedProject).filter({it.sampleDataAvailable > 0 }).ifPresent({
+            selectedSummary = Optional.ofNullable(viewModel.selectedProject)
+            selectedSummary.filter({it.sampleDataAvailable.passingSamples > 0 }).ifPresent({
                 String projectCode = it.getCode()
                 downloadProjectController.downloadProject(projectCode)
             })
         } catch (IllegalArgumentException illegalArgument) {
-            notificationService.publishFailure("Manifest Download failed due to: ${illegalArgument.getMessage()}")
-        } catch (Exception ignored) {
+            notificationService.publishFailure("Manifest Download failed for project ${viewModel.selectedProject.getCode()}. ${Constants.CONTACT_HELPDESK}")
```

Comment:
Thank you for tackling this! What do you think about moving the evaluation `${viewModel.selectedProject.getCode()}` out of here into an optional so you can ensure that it is what was used above?
```suggestion
            String projectCode = selectedSummary.map(ProjectSummary::getCode).orElse("No project selected")
            notificationService.publishFailure("Manifest Download failed for project ${projectCode}. ${Constants.CONTACT_HELPDESK}")
```